builder/dockerfile/internals_windows.go

```diff
@@ -0,0 +1,392 @@
+package dockerfile
+
+// internals for handling commands. Covers many areas and a lot of
+// non-contiguous functionality. Please read the comments.
+
+import (
+	"os"
+	"path/filepath"
+	"strconv"
+	"unsafe"
+
+	"github.com/Microsoft/go-winio"
+	"github.com/docker/docker/pkg/fileutils"
+	"github.com/docker/docker/pkg/idtools"
+	"github.com/docker/docker/pkg/ioutils"
+	"github.com/docker/docker/pkg/symlink"
+	"github.com/docker/docker/pkg/system"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+func getAccountIdentity(accountName string, ctrRootPath string) (idtools.Identity, error) {
+
+	sid, _, accType, err := windows.LookupSID("", accountName)
+
+	if err == nil && (accType == windows.SidTypeAlias || accType == windows.SidTypeWellKnownGroup) {
+		accountSid, err := sid.String()
+
+		if err != nil {
+			return idtools.Identity{IdType: idtools.TypeIDSID, IdSid: ""}, errors.Wrapf(err, "error converting SID to string")
+		}
+
+		return idtools.Identity{IdType: idtools.TypeIDSID, IdSid: accountSid}, nil
+	}
+
+	if accountName == "ContainerAdministrator" {
+		return idtools.Identity{IdType: idtools.TypeIDSID, IdSid: "S-1-5-93-2-1"}, nil
+
+	} else if accountName == "ContainerUser" {
```

Comment:
These comparisons for `ContainerAdministrator` and `ContainerUser` should be case-insensitive.